lib/discover.js

```diff
@@ -217,31 +228,27 @@ function Discover (options, callback) {
 
 			return false;
 		}
-
-		self.leadershipElector.start(self, function (err) {
+		
+		self.broadcast.start(function (err) {
```

Comment:
Need to keep this section as it was, as in my use case where I passed my own instantiation, I have no reference to the discover object yet.
start was my hook into receiving it.

auto connect false would open door to inject it myself, but say I instantiate BasicLeadership myself and extend it, that code depends on having a reference to this.discover.

That needs to be maintained.

Otherwise the resolveLeadership needs to inject the discover reference.

I think if you revert the start changes here, so that .start() can still be used to "intercept" start, then this would not be an API breaking change and overall looks good to me.

Reply:
There is a `started` event that is emitted which is bound to the start method of the Leadership instance which receives the instance of node-discover. So, unless I'm misunderstanding, I think this is covered without having to revert this change.

Reply:
slight behavioral change in that it no longer allows controlling the start, but I'm ok with that as i can refactor my code to account for it, since the start param can control delaying start too.

Reply:
Oh I think I understand now. You were controlling the start by waiting to call the callback that was passed to the start method. I can't think of a clever way to handle this without reverting the change. Thank you for refactoring your code to account for the change.